src/bgw/job.c

```diff
@@ -603,63 +578,44 @@ ts_bgw_job_validate_job_owner(Oid owner, JobType type)
 		ReleaseSysCache(role_tup);
 		ereport(ERROR,
 				(errcode(ERRCODE_INVALID_AUTHORIZATION_SPECIFICATION),
-				 errmsg("permission denied to start %s background process as role \"%s\"",
-						job_type_names[type],
+				 errmsg("permission denied to start background process as role \"%s\"",
 						rform->rolname.data),
 				 errhint("Hypertable owner must have LOGIN permission to run background tasks.")));
 	}
 	ReleaseSysCache(role_tup);
 }
 
+static bool
+is_telemetry_job(BgwJob *job)
+{
+	return namestrcmp(&job->fd.proc_name, "policy_telemetry_proc") == 0;
```

Comment:
This should probably include namespace/schema as well and maybe check that this function is part of our extension.

Reply:
we use this just as marker for the telemetry job we don't execute the procedure though but call a hardcoded internal telemetry c function so there are no security implications, otherwise people could replace the telemetry procedure and get their own functions executed

Reply:
Ok, so then there is at least not a security issue, but this is still quite obscure. Could there be some other way to indicate that this is the telemetry job that is less obscure? Set this column to NULL and use the `application_name` column?

Reply:
well I disagree proc_name/proc_schema determine what gets executed just that we dont do lookup here but go straight to a c function